### PR TITLE
[LTC] Replace XLA_VALUE_METRIC with TORCH_LAZY_VALUE_METRIC

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This pull request replaces all XLA_VALUE_METRIC in pytorch/xla with upstream
TORCH_LAZY_VALUE_METRIC.

This is paired with pytorch/pytorch#89606.

Test Plan:
PJRT_DEVICE=CPU python test/test_metrics.py

### CODE REVIEW

File: test/test_metrics.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def test_short_metrics_fallback_counter(self):
   def test_metrics_report(self):
     # TODO(jwtan): Add test to cover TrimIrGraph, SyncTensorsToData, TransferToServerAsync, IrValueTensorToXlaData
     xla_device = xm.xla_device()
-    t1 = torch.tensor(1456, device=xla_device)
+    t1 = torch.tensor(2077, device=xla_device)
```
Comment (JackCaoG) [inline test/test_metrics.py:97]: why this change?

Comment (alanwaketan) [reply test/test_metrics.py:97]: It looks like somehow we cache the device data in the server? If we reuse the same value for the scalar tensor, `TransferToServerTime` will be omitted in some of the tests.

Comment (JackCaoG) [reply test/test_metrics.py:97]: oh right.

**Review by JackCaoG — APPROVED** [review]
Thanks!
